### PR TITLE
Added MosaicML Provider Documentation

### PR DESCRIPTION
**Pull request by arpitjasa-db**
Added docs

### CODE REVIEW

File: docs/source/gateway/index.rst [hunk 1]
```diff
@@ -390,6 +390,10 @@ below can be used as a helpful guide when configuring a given route for any newl
      - OpenAI
      - gpt-3.5-turbo, gpt-4
      - Yes
+   * - llm/v1/completions
+     - MosaicML
+     - mpt-7b-instruct, mpt-30b-instruct, llama2-70b-chat†
```
Comment (arpitjasa-db) [inline docs/source/gateway/index.rst:396]: Not sure if † is the best symbol for this blurb

**Review by vladimirk-db — APPROVED** [review]
LGTM!

**Review by lupesko — APPROVED** [review]
shipit! 🛳️

File: docs/source/gateway/index.rst [hunk 2]
```diff
@@ -446,6 +458,7 @@ below can be used as a helpful guide when configuring a given route for any newl
      - Databricks Model Serving
      - Endpoints with compatible schemas
      - Yes
+† Llama 2 is licensed under the LLAMA 2 Community License, Copyright © Meta Platforms, Inc. All Rights Reserved.
```
Comment (vladimirk-db) [inline docs/source/gateway/index.rst]:
```suggestion
† Llama 2 is licensed under the [LLAMA 2 Community License](https://ai.meta.com/llama/license/), Copyright © Meta Platforms, Inc. All Rights Reserved. 
```

File: docs/source/gateway/index.rst [hunk 3]
```diff
@@ -410,6 +414,10 @@ below can be used as a helpful guide when configuring a given route for any newl
      - OpenAI
      - gpt-3.5-turbo, gpt-4
      - Yes
+   * - llm/v1/chat
```
Comment (vladimirk-db) [inline docs/source/gateway/index.rst:418]: Can you please also add a blurb (rst Note) somewhere:
"Customers are responsible for ensuring compliance with applicable model licenses."

Probably up top before we list all the modes.

**Pull request merged**